temporian/implementation/numpy/data/event.py

```diff
@@ -25,81 +25,148 @@
     int: DType.INT64,
 }
 
+DTYPE_MAPPING = {
+    np.float64: DType.FLOAT64,
+    np.float32: DType.FLOAT32,
+    np.int64: DType.INT64,
+    np.int32: DType.INT32,
+    np.str_: DType.STRING,
+    np.string_: DType.STRING,
+    np.bool_: DType.BOOLEAN,
+}
+DTYPE_REVERSE_MAPPING = {
+    DType.FLOAT64: np.float64,
+    DType.FLOAT32: np.float32,
+    DType.INT64: np.int64,
+    DType.INT32: np.int32,
+    DType.STRING: np.str_,
+    DType.BOOLEAN: np.bool_,
+}
+
+
+@dataclass
+class IndexData:
+    features: List[np.ndarray]
+    timestamps: np.ndarray
+
+    def __init__(
+        self,
+        features: Union[np.ndarray, List[np.ndarray]],
+        timestamps: np.ndarray,
+    ) -> None:
+        if isinstance(features, np.ndarray):
+            features = [features]
+
+        self.features = features
+        self.timestamps = timestamps
+
+    def __len__(self) -> int:
+        return len(self.timestamps)
+
 
 class NumpyEvent:
+    # tolerance levels for comparing float features in __eq__
+    rtol: float = 1e-9
+    atol: float = 1e-9
+
     def __init__(
         self,
-        data: Dict[Tuple, List[NumpyFeature]],
-        sampling: NumpySampling,
+        data: Dict[Tuple, IndexData],
+        feature_names: Union[str, List[str]],
```

Comment:
Same here (data and feature_names). I would only allow Lists.
Overloading argument types make the code harder to read.